src/lib/selections.ts

```diff
@@ -527,17 +550,16 @@ export function getSelectionType(
   selection?: Selections
 ): ResolvedSelectionType[] {
   if (!selection) return []
-  const extrudableCount = selection.codeBasedSelections.filter((_, i) => {
-    const singleSelection = {
-      ...selection,
-      codeBasedSelections: [selection.codeBasedSelections[i]],
-    }
-    return canExtrudeSelectionItem(singleSelection, 0)
-  }).length
-
-  return extrudableCount === selection.codeBasedSelections.length
-    ? [['extrude-wall', extrudableCount]]
-    : [['other', selection.codeBasedSelections.length]]
+  const selectionsWithArtifacts = selection.graphSelections.filter(
+    (s) => !!s.artifact
+  )
+  const firstSelection = selectionsWithArtifacts[0]
+  const firstSelectionType = firstSelection?.artifact?.type
+  if (!firstSelectionType) return []
+  const selectionsWithSameType = selectionsWithArtifacts.filter(
+    (s) => s.artifact?.type === firstSelection.artifact?.type
+  )
+  return [[firstSelectionType, selectionsWithSameType.length]]
```

Comment:
@franknoirot I wasn't completely sure what the intention with the type `ResolvedSelectionType = [Artifact['type'] | 'other', number]` was

what I've got here is working, but maybe it's supposed to group the selections by type and return the how many of each type?

Reply:
Ah yeah exactly. That format of selection is used both in
1. [`canSubmitSelectionArg`](https://github.com/KittyCAD/modeling-app/blob/main/src/lib/selections.ts#L559) to let us restrict the user to have, for example, exactly two faces selected in the future for a sweep user flow
2. [`getSelectionType`](https://github.com/KittyCAD/modeling-app/blob/main/src/lib/selections.ts#L526) and by extension [`getSelectionTypeDisplayText`](https://github.com/KittyCAD/modeling-app/blob/main/src/lib/selections.ts#L543) to display the current selection in the command bar (both on the selection arg step and in the little arg summary button in the command bar header).

I'll take a look at those parts to see if everything works similar to before